### PR TITLE
FPL-37: Added enter respondents functionality to configuration

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPL-37

### Change description ###

Added Respondent complex type to configruation spreadsheet that consists of a respondentType and a collection of respondentType giving the following data structure:
```
data: { 
    respondents: {
        firstResponent: {}
        additional: { 
            0: {}, 
            1: {},
            ...
        }
    }
}
``` 

Created reusable tests that take in an id value to identify which respondent is being entered in the UI.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by swalker125** [comment]
screenshot_2018-10-19 core case data hmcts
screenshot_2018-10-19 core case data hmcts 1
screenshot_2018-10-19 core case data hmcts 2

File: e2e/fixtures/respondent.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  name: 'Joe Bloggs',
+  dob: {
+    day: '01',
+    month: '01',
+    year: '1980',
+  },
+  gender: 'male',
```
Comment (kurt-uk) [inline e2e/fixtures/respondent.js]: this should be capital M if youre passing it into the dropdown field, `Male`

File: e2e/pages/enterRespondents/enterRespondents.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      },
+      abilityToTakePartInProceedings: (option) => {
+        return {
+          option: `#respondent_${id}_abilityToTakePartInProceedings-${option}`,
+          reason: `#respondent_${id}_abilityToTakePartInProceedingsReason`,
+        };
+      },
+    };
+  },
+  addOtherRespondent: 'Add new',
```
Comment (kurt-uk) [inline e2e/pages/enterRespondents/enterRespondents.js]: just `addRespondent`?

File: e2e/fixtures/respondent.js (hunk 1 above)
Comment (swalker125) [reply e2e/fixtures/respondent.js]: just refactored to use the same GenderList as FPL-39, making the changes in the tests now 👍

File: e2e/pages/enterRespondents/enterRespondents.js (hunk 2 above)
Comment (swalker125) [reply e2e/pages/enterRespondents/enterRespondents.js]: 👍

File: e2e/fixtures/respondent.js (hunk 1 above)
Comment (kurt-uk) [reply e2e/fixtures/respondent.js]: good boy

Comment (kurt-uk) [reply e2e/fixtures/respondent.js]: *good male

Comment (swalker125) [reply e2e/fixtures/respondent.js]: male or Male doesn't appear to make a difference

**Comment by geek-soft** [comment]
I think we should name case field `respondents` not `respondent`

**Comment by swalker125** [comment]
changed @geek-soft I did mean it to be respondents originally!

File: e2e/fixtures/respondent.js (hunk 1 above)
Comment (swalker125) [reply e2e/fixtures/respondent.js]: It doesn't or my tests would fail and say the option doesn't exist. Changed to M anyway.

**Pull request merged**